### PR TITLE
[Admin][Order] Show page

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | N/A
| License         | MIT

image

### CODE REVIEW

**Review by mpysiak — DISMISSED** [review]

File: src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/items/body/distributed_order_discount.html.twig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% import '@SyliusAdmin/shared/helper/money.html.twig' as _money %}
+
+{% set orderPromotionAdjustment = constant('Sylius\\Component\\Core\\Model\\AdjustmentInterface::ORDER_PROMOTION_ADJUSTMENT') %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/items/body/distributed_order_discount.html.twig]: Use snake_case notation for Twig variables

File: src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/items/body/subtotal.html.twig [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{% import '@SyliusAdmin/shared/helper/money.html.twig' as _money %}
+
+{% set orderPromotionAdjustment = constant('Sylius\\Component\\Core\\Model\\AdjustmentInterface::ORDER_PROMOTION_ADJUSTMENT') %}
+{% set unitPromotionAdjustment = constant('Sylius\\Component\\Core\\Model\\AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT') %}
+
+{% set order = hookable_metadata.context.resource %}
+{% set item = hookable_metadata.context.item %}
+{% set aggregatedUnitPromotionAdjustments = item.getAdjustmentsTotalRecursively(unitPromotionAdjustment) + item.getAdjustmentsTotalRecursively(orderPromotionAdjustment) %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/items/body/subtotal.html.twig]: Use snake_case also here and in other templates

File: src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/items/foot.html.twig [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% import '@SyliusAdmin/shared/helper/money.html.twig' as _money %}
+
+{% set order = hookable_metadata.context.resource %}
+
+<tfoot>
+    <tr>
+        <td class="h3 mb-0" colspan="7">{{ 'sylius.ui.total'|trans }}</td>
+        <td class="text-end h3 mb-0">{{ _money.format(order.taxTotal, order.currencyCode) }}</td>
+        <td class="text-end h3 mb-0" {{ sylius_test_html_attribute('items-total') }}>{{ _money.format(order.itemsTotal, order.currencyCode) }}</td>
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/items/foot.html.twig]: Hook? To be consistent with head and body of this items table? 🤔

File: features/admin/order/managing_orders/order_details/seeing_order_with_items.feature [hunk 4]
```diff
@@ -12,10 +12,11 @@ Feature: Seeing an order with its items
         And the store allows paying with "Cash on Delivery"
         And there is a customer "lucy@teamlucifer.com" that placed an order "#00000666"
         And the customer bought an "Angel T-Shirt" and an "Angel Mug"
+        And the customer set the billing address as "Mike Ross", "350 5th Ave", "10118", "New York", "United States"
         And the customer chose "Free" shipping method to "United States" with "Cash on Delivery" payment
```
Comment (Rafikooo) [inline features/admin/order/managing_orders/order_details/seeing_order_with_items.feature:16]: I don't understand why we first purchase the products and then set up billing and shipping.

File: src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/items/body/subtotal.html.twig [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{% import '@SyliusAdmin/shared/helper/money.html.twig' as _money %}
```
Comment (Rafikooo) [inline src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/items/body/subtotal.html.twig:1]:
```suggestion
{% import '@SyliusAdmin/shared/helper/money.html.twig' as money %}
```
As far as I know, we agreed on a convention without using the `_` prefix.

File: src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/summary/shipping_total.html.twig [hunk 6]
```diff
@@ -1,17 +1,37 @@
...
+                {% for label, amount in order_shipping_promotions %}
+                    <div>{{ label }}: <span {{ sylius_test_html_attribute('shipping-promotion-discount') }}>{{ _money.format(amount, order.currencyCode) }}</span></div>
+                {% endfor %}
             </div>
         </div>
     </td>
-    <td class="text-end">{{ _money.format(order.shippingTotal, order.currencyCode) }}</td>
+    <td class="text-end" {{ sylius_test_html_attribute('shipping-total') }}>{{ _money.format(order.shippingTotal, order.currencyCode) }}</td>
 </tr>
+
```
Comment (Rafikooo) [inline src/Sylius/Bundle/AdminBundle/templates/order/show/content/sections/summary/shipping_total.html.twig:37]:
```suggestion
```

**Review by Rafikooo — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Pull request merged**